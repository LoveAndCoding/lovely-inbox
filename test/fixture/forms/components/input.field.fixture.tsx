import * as React from "react";
import * as TestRenderer from "react-test-renderer";

import InputField, {
	InputFieldSizes,
} from "src/ui/forms/components/input.field";
import { UNSAFE_resetUniqueIdCounters } from "src/ui/layout/uid.factory";

describe("<InputField>", () => {
	beforeEach(() => {
		// We need to reset the counters so our snapshot IDs are predictable
		UNSAFE_resetUniqueIdCounters();
	});

	test("Input type 'text' renders correctly", () => {
		// Arrange

		// Act
		const tree = TestRenderer.create(
			<InputField label="Text Test" type="text" />,
		).toJSON();

		// Assert
		expect(tree).toMatchSnapshot();
	});

	test("Input type 'text' with 'min/maxLength' renders correctly", () => {
		// Arrange

		// Act
		const tree = TestRenderer.create(
			<InputField
				label="Text Attr Test"
				maxLength={10}
				minLength={1}
				type="text"
			/>,
		).toJSON();

		// Assert
		expect(tree).toMatchSnapshot();
	});

	test("Input type 'number' renders correctly", () => {
		// Arrange

		// Act
		const tree = TestRenderer.create(
			<InputField label="Number Test" type="number" />,
		).toJSON();

		// Assert
		expect(tree).toMatchSnapshot();
	});

	test("Input type 'number' with 'min'/'max'/'step' renders correctly", () => {
		// Arrange

		// Act
		const tree = TestRenderer.create(
			<InputField
				label="Number Attr Test"
				max={10}
				min={1}
				step={0.1}
				type="number"
			/>,
		).toJSON();

		// Assert
		expect(tree).toMatchSnapshot();
	});

	test("Input type 'date' renders correctly", () => {
		// Arrange

		// Act
		const tree = TestRenderer.create(
			<InputField label="Date Test" type="date" />,
		).toJSON();

		// Assert
		expect(tree).toMatchSnapshot();
	});

	test("Input attr 'required' renders correctly", () => {
		// Arrange

		// Act
		const tree = TestRenderer.create(
			<InputField label="Required Attr Test" required type="number" />,
		).toJSON();

		// Assert
		expect(tree).toMatchSnapshot();
	});
});
