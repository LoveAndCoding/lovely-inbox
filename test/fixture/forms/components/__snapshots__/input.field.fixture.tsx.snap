// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField> Input attr 'required' renders correctly 1`] = `
Array [
  <label
    className="label_5g4e5d"
    htmlFor="#text-input-1"
  >
    Required Attr Test
  </label>,
  <input
    className="input_1us7r7i-o_O-medium_1jznm8k"
    id="text-input-1"
    max={null}
    maxLength={null}
    min={null}
    minLength={null}
    onChange={[Function]}
    required={true}
    step={null}
    type="number"
  />,
]
`;

exports[`<InputField> Input type 'date' renders correctly 1`] = `
Array [
  <label
    className="label_5g4e5d"
    htmlFor="#text-input-1"
  >
    Date Test
  </label>,
  <input
    className="input_1us7r7i-o_O-medium_1jznm8k"
    id="text-input-1"
    max={null}
    maxLength={null}
    min={null}
    minLength={null}
    onChange={[Function]}
    required={false}
    step={null}
    type="date"
  />,
]
`;

exports[`<InputField> Input type 'number' renders correctly 1`] = `
Array [
  <label
    className="label_5g4e5d"
    htmlFor="#text-input-1"
  >
    Number Test
  </label>,
  <input
    className="input_1us7r7i-o_O-medium_1jznm8k"
    id="text-input-1"
    max={null}
    maxLength={null}
    min={null}
    minLength={null}
    onChange={[Function]}
    required={false}
    step={null}
    type="number"
  />,
]
`;

exports[`<InputField> Input type 'number' with 'min'/'max'/'step' renders correctly 1`] = `
Array [
  <label
    className="label_5g4e5d"
    htmlFor="#text-input-1"
  >
    Number Attr Test
  </label>,
  <input
    className="input_1us7r7i-o_O-medium_1jznm8k"
    id="text-input-1"
    max={10}
    maxLength={null}
    min={1}
    minLength={null}
    onChange={[Function]}
    required={false}
    step={0.1}
    type="number"
  />,
]
`;

exports[`<InputField> Input type 'text' renders correctly 1`] = `
Array [
  <label
    className="label_5g4e5d"
    htmlFor="#text-input-1"
  >
    Text Test
  </label>,
  <input
    className="input_1us7r7i-o_O-medium_1jznm8k"
    id="text-input-1"
    max={null}
    maxLength={null}
    min={null}
    minLength={null}
    onChange={[Function]}
    required={false}
    step={null}
    type="text"
  />,
]
`;

exports[`<InputField> Input type 'text' with 'min/maxLength' renders correctly 1`] = `
Array [
  <label
    className="label_5g4e5d"
    htmlFor="#text-input-1"
  >
    Text Attr Test
  </label>,
  <input
    className="input_1us7r7i-o_O-medium_1jznm8k"
    id="text-input-1"
    max={null}
    maxLength={10}
    min={null}
    minLength={1}
    onChange={[Function]}
    required={false}
    step={null}
    type="text"
  />,
]
`;
